src/components/DisplayProfile.jsx

```diff
@@ -1,23 +1,27 @@
 import { useEnsAddress, useNativeTransactions } from "react-moralis";
-import  getProfiles  from '@/lens/get-profiles.js'
+import getProfiles from "@/lens/get-profiles.js";
+import doesFollowFunc from '@/lens/does-follow'
 import { useState } from "react";
-
+import FollowBtn from "./FollowBtn";
+import UnfollowBtn from "./UnfollowBtn";
 
 export default function DisplayProfile(props) {
-  const [ profile, setProfile] = useState()
+  const [profile, setProfile] = useState();
+  const [doesFollow, setDoesFollow] = useState();
 
   const { name } = useEnsAddress(props.address);
   const { data: Transactions, error } = useNativeTransactions({
     address: props?.address,
   });
 
   const getProfile = async () => {
-    let { profiles } = await getProfiles({ownedBy: props?.address});
-    setProfile(profiles.items[0])
-  }
-
-  getProfile()
+    let { profiles } = await getProfiles({ ownedBy: props?.address });
+    setProfile(profiles.items[0]);
+    setDoesFollow(await doesFollowFunc(profile?.id))
```

Comment:
This trigger a lot of error because the first few time doesFollowFunc is called, the state profile is not updated yet. So this trigger the error you see in the video I sent in the discord.

You should use the result from your previous query, since you're waiting for it, you're guarantee to have the id